syntax = "proto3";

package ServerRpc.auth;

message userTokenReq {
  string secret = 1;
  int32  platformID = 2;
  string userID = 3;
  string password = 4;
}

message userTokenResp {
  string token = 2;
  int64  expireTimeSeconds = 3;
}

message forceLogoutReq {
  int32  platformID = 1;
  string userID = 2;
}

message forceLogoutResp {
}

message parseTokenReq{
  string token = 1;
}

message parseTokenResp{
  string userID = 1;
  int32 platformID = 2;
  int64 expireTimeSeconds = 4;
}

service Auth {
  //生成token
  rpc userToken(userTokenReq) returns(userTokenResp);
  //强制退出登录
  rpc forceLogout(forceLogoutReq) returns(forceLogoutResp);
  //解析token
  rpc parseToken(parseTokenReq)returns(parseTokenResp);
}